src/windows.rs

```diff
@@ -25,23 +37,34 @@ pub(super) fn open_browser_internal(
             }
 
             trace!("trying to figure out default browser command");
-            let mut cmdline_u16 = [0_u16; 512];
-            let pwstr = PWSTR::from_raw((&mut cmdline_u16) as *mut u16);
             let cmdline = unsafe {
-                let mut line_len: u32 = 512;
-                AssocQueryStringW(
+                const BUF_SIZE: usize = 512;
+                let mut cmdline_u16 = [0_u16; BUF_SIZE];
+                let mut line_len = BUF_SIZE as u32;
+                if AssocQueryStringW(
                     ASSOCF_IS_PROTOCOL,
                     ASSOCSTR_COMMAND,
-                    w!("http"),
-                    PCWSTR::null(),
-                    pwstr,
-                    &mut line_len as *mut u32,
-                )
-                .map_err(err_other)?;
+                    [0x68, 0x74, 0x74, 0x70, 0x0].as_ptr(), // http\0
```

Comment:
Wouldn't a `b"http\0"` string have worked here... without needing a comment?

Reply:
No, that's a utf-8 string, this needs to be utf-16.